packages/components/src/internal/url/AppLink.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, memo, PropsWithChildren, StyleHTMLAttributes, MouseEvent } from 'react';
+
+import { Link } from 'react-router-dom';
+
+import { ActionURL } from '@labkey/api';
+
+import { AppURL } from './AppURL';
+import { isTestEnv } from '../util/utils';
+
+const TARGET_BLANK = '_blank';
+const URL_REL = 'noopener noreferrer';
+
+/**
+ * If the given href points to our current container & product, return the react router path, in all other cases
+ * return undefined.
+ * @param href
+ */
+export function parseAppPath(href: string): string | undefined {
+    if (href.startsWith('#')) {
+        // This typically happens when a URL Mapper uses AppURl.create().toHref()
+        return href.replace('#', '');
+    }
+
+    const contextPath = ActionURL.getContextPath();
+    const container = ActionURL.getContainer();
+    const controller = ActionURL.getController();
+    const action = ActionURL.getAction();
+    const baseAppUrl = `${contextPath}${container}/${controller}-${action}.view#`;
```

Comment:
We will want to leverage `ActionURL` to generate the `baseAppUrl` so we are getting consistent container path placement in url.